fairseq/sequence_generator.py

```diff
@@ -375,6 +397,8 @@ def _generate(
                 tokens[:, : step + 1],
                 original_batch_idxs,
             )
+            # update states for next step
+            self.search.update_negative_constraints(cand_indices, cand_beams)
```

Comment:
Same, this can be moved to the end of `LexicallyConstrainedBeamSearch::step`

Reply:
Done.